website/blog/2021-03-11-jest-27.md

```diff
@@ -0,0 +1,80 @@
+For this reason, we are **changing the default test environment** from `"jsdom"` to `"node"`. If you are affected by this change because you use DOM APIs and do not have the test environment explicitly configured, you should be receiving an error when e.g. `document` is accessed, and you can configure `"testEnvironment": "jsdom"` or use [per-file environment configuration](/docs/configuration#testenvironment-string) to resolve this.  
+For mixed projects where some tests require a DOM environment but others don't, we recommend using the fast `"node"` environment by default and declaring exactly those tests that need the DOM using [docblocks](/docs/configuration#testenvironment-string).  
+In the next major, we plan to also eliminate `jest-jasmine2` and `jest-environment-jsdom` from the Jest dependency tree and require them to be installed explicitly, so that many users can profit from a smaller install size with less clutter that they don't need.
+
+Another default that we are changing affects Fake Timers aka [Timer Mocks](/docs/timer-mocks). We introduced an opt-in "modern" implementation of Fake Timers in Jest 26 accessed transparently through the same API, but with much more comprehensive mocking, such as for `Date` and `queueMicrotask`.  
+This **modern fake timers implementation will now be the default**. If you are among the unlucky few who are affected by the subtle implementation differences too heavily to migrate, you can get back the old implementation using `jest.useFakeTimers("legacy")` or, if you are enabling fake timers globally via [configuration](/docs/configuration#timers-string), `"timers": "legacy"`.
+
+## Features coming with breaking changes
+
+We introduced a few more small breaking changes to help you avoid mistakes by disallowing some things that can easily happen unintentionally:
+
+- The same `done` test callback may not be called more than once,
+- calling `done` and returning a Promise may not be combined,
+- a `describe` block must not return anything,
+
+and we [made some TypeScript types stricter](https://github.com/facebook/jest/pull/10512).
+
+Modules used in the following configuration options are now transformed like the rest of your code, which may be breaking if you relied on them being loaded as-is:
+
+- `testEnvironment`
+- `runner`
+- `testRunner`
+- `snapshotResolver` <!-- TODO pending merge https://github.com/facebook/jest/pull/8829 -->
+
+## Miscellaneous breaking changes
+
+We removed some long-deprecated functions:
+
+- `jest.addMatchers` (use `expect.extend` instead)
+- `jest.resetModuleRegistry` (use `jest.resetModules` instead)
+- `jest.runTimersToTime` (use `jest.advanceTimersByTime` instead)
+
+A lot of Jest's packages have been migrated to use ESM exports, so if you consume e.g. `pretty-format` directly, you may need to adjust your import to a `default` import.
```

Comment:
Should probably mentions it's still CJS via Babel's transform. It's not _native_ ESM